src/sidebar/markdown-commands.js

```diff
@@ -239,18 +241,26 @@ function transformLines(state, start, end, callback) {
  * Toggle Markdown-style formatting around a block of text.
  *
  * @param {EditorState} state - The current state of the input field.
- * @param {string} prefix - The prefix to add or remove before each line
- *                          of the selection.
+ * @param {string|((lineNum: number) => string)} prefixBuilder - The prefix
+ *        to add or remove before each line of the selection, or a callback
+ *        which returns said prefix for a specific line number (useful for
+ *        multi-line selections that require different prefixes per line).
  * @return {EditorState} - The new state of the input field.
  */
-export function toggleBlockStyle(state, prefix) {
+export function toggleBlockStyle(state, prefixBuilder) {
```

Comment:
If passing a string for prefix is still supported / the most common situation, I'd leave the name of this as `prefix`.

Reply:
That was actually my first approach, but then I had to come up with another name for when you have resolved it 😅 

Will change it back.